### PR TITLE
Switch spec and implementation to use strict booleans

### PR DESCRIPTION
**Pull request by michaelpj**
Thought I'd see how much effort this was to implement, the answer is not much. All my test output gets much simpler :)

Also this fixes the examples in the spec, where `case` needs to be lazy otherwise the examples are always `error`.

We shouldn't merge this until we've decided it's actually what we want to do, though.

### CODE REVIEW

File: core-to-plc/src/Language/Plutus/CoreToPLC.hs [hunk 1]
```diff
@@ -451,41 +451,41 @@ isPrimitiveDataCon dc = dc == GHC.intDataCon
...
 convOrdMethod :: (Converting m) => GHC.Name -> m PLCExpr
 convOrdMethod name
     -- only this one has a name defined in the lib??
-    | name == GHC.geName = mkIntRel PLC.GreaterThanEqInteger
-    | GHC.getOccString name == ">" = mkIntRel PLC.GreaterThanInteger
-    | GHC.getOccString name == "<=" = mkIntRel PLC.LessThanEqInteger
-    | GHC.getOccString name == "<" = mkIntRel PLC.LessThanInteger
+    | name == GHC.geName = pure $ mkIntRel PLC.GreaterThanEqInteger
+    | GHC.getOccString name == ">" = pure $ mkIntRel PLC.GreaterThanInteger
+    | GHC.getOccString name == "<=" = pure $ mkIntRel PLC.LessThanEqInteger
```
Comment (effectfully) [inline core-to-plc/src/Language/Plutus/CoreToPLC.hs]: Writing something like

```haskell
case GHC.getOccString name of
    ">"  -> ...
    "<=" -> ...
```

looks slightly nicer to my taste, but is not important.

File: core-to-plc/src/Language/Plutus/CoreToPLC.hs [hunk 2]
```diff
@@ -451,41 +451,41 @@ isPrimitiveDataCon dc = dc == GHC.intDataCon
...
 
 convEqMethod :: (Converting m) => GHC.Name -> m PLCExpr
 convEqMethod name
-    | name == GHC.eqName = mkIntRel PLC.EqInteger
+    | name == GHC.eqName = pure $ mkIntRel PLC.EqInteger
     | otherwise = throwSd UnsupportedError $ "Eq method:" GHC.<+> GHC.ppr name
 
 convOrdMethod :: (Converting m) => GHC.Name -> m PLCExpr
 convOrdMethod name
     -- only this one has a name defined in the lib??
```
Comment (effectfully) [inline core-to-plc/src/Language/Plutus/CoreToPLC.hs]: I think I would consistently use `GHC.getOccString` then.

File: core-to-plc/test/ifThenElseApply.plc.golden [hunk 3]
```diff
@@ -1 +1 @@
-(con 64 ! 2)
\ No newline at end of file
+(con 64 ! 1)
```
Comment (effectfully) [inline core-to-plc/test/ifThenElseApply.plc.golden]: A different result?

File: core-to-plc/src/Language/Plutus/CoreToPLC.hs (hunk 1 above)
Comment (michaelpj) [reply core-to-plc/src/Language/Plutus/CoreToPLC.hs]: All the cases are a bit special atm - I'll come back to this and the other style suggestions on other bits of code in a follow up.

File: core-to-plc/src/Language/Plutus/CoreToPLC.hs (hunk 2 above)
Comment (michaelpj) [reply core-to-plc/src/Language/Plutus/CoreToPLC.hs]: I'm still hoping I'll find the other names defined somewhere... I'd rather use the one from GHC if possible.

File: core-to-plc/test/ifThenElseApply.plc.golden (hunk 3 above)
Comment (michaelpj) [reply core-to-plc/test/ifThenElseApply.plc.golden]: :scream: looking into it

**Comment by michaelpj** [comment]
Fixed the boolean mismatch by special-casing Haskell's Bool and swapping the case branches in that case.

**Pull request merged**